IntelliTect.TestTools.TestFramework/BlockData.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace IntelliTect.TestTools.TestFramework;
+
+public interface IBlockData
+{
+    public List<KeyValuePair<Type, object?>> Data { get; }
+}
+
+public class BlockData<T1, T2>(T1 data1, T2 data2) : IBlockData
+{
+    static BlockData()
+    {
+        ValidateData.ValidateUniqueTypes(typeof(T1), typeof(T2));
+    }
+
+    public List<KeyValuePair<Type, object?>> Data { get; } = 
+    [
+        new KeyValuePair<Type, object?>(typeof(T1), data1),
+        new KeyValuePair<Type, object?>(typeof(T2), data2)
+    ];
```

Comment:
Just as a thought to make the API a bit cleaner. It also avoids a boxing operation for anyone wanting to view the data.

```suggestion
    public T1 Data1 => data1;
    public T2 Data2 => data2;
    

    IBlockData.List<KeyValuePair<Type, object?>> Data { get; } = 
    [
        new KeyValuePair<Type, object?>(typeof(T1), data1),
        new KeyValuePair<Type, object?>(typeof(T2), data2)
    ];
```